crates/runtime/src/dataconnector/graphql.rs

```diff
@@ -187,9 +221,45 @@ impl GraphQLClient {
             });
         }
 
-        for key in self.json_path.split('.') {
-            response = response[key].take();
-        }
+        let pointer = format!("/{}", self.json_path.replace(".", "/"));
+
+        let next_cursor = match (&self.paginated_resource_name, limit_reached) {
+            (Some(resource), false) => {
+                let page_info = {
+                    // trim path to paginated resource
+                    let pattern = format!(r"^(.*?{resource})");
+                    let regex = Regex::new(pattern.as_str()).unwrap();
+                    let captures = regex.captures(&pointer);
+                    let path = captures.map_or(None, |c| {
+                        c.get(1).map(|m| m.as_str().to_owned() + "/pageInfo")
+                    });
```

Comment:
This also can be done on client initialization, as it depends only on json_path and paginated_resource_name